src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/yflow/update/actions/ValidateYFlowAction.java

```diff
@@ -127,10 +127,20 @@ protected Optional<Message> performWithResponse(State from, State to, Event even
                     format("Unable to map provided sub-flows set onto existing y-flow %s", yFlowId));
         }
 
-        YSubFlow subFlow = yFlow.getSubFlows().stream().findAny()
-                .orElseThrow(() -> new FlowProcessingException(ErrorType.DATA_INVALID,
-                        format("No sub-flows of the y-flow %s were found", yFlowId)));
-        stateMachine.setMainAffinityFlowId(subFlow.getFlow().getAffinityGroupId());
+        YSubFlow subFlow = yFlow.getSubFlows().stream()
+                .filter(p -> !p.getEndpointSwitchId().equals(yFlow.getSharedEndpoint().getSwitchId()))
+                .findAny()
+                .orElse(null);
+        if (subFlow != null) {
+            stateMachine.setMainAffinityFlowId(subFlow.getFlow().getAffinityGroupId());
+        } else {
+            // if there is no multi switch flows we have to use one switch flow
+            YSubFlow oneSwitchSubFlow = yFlow.getSubFlows().stream()
+                    .findAny()
+                    .orElseThrow(() -> new FlowProcessingException(ErrorType.DATA_INVALID,
+                    format("No sub-flows of the y-flow %s were found", yFlowId)));
+            stateMachine.setMainAffinityFlowId(oneSwitchSubFlow.getSubFlowId());
+        }
```

Comment:
I'd move this logic to YFlow and YSubFlow classes. Or even overload the getSubFlows() method. So this part of validator would look like
```
if (yFlow.getSubFlows(/*sameSwitchOnly=*/true)).empty()) {
    flowId = yFlow.getSubFlows().get(0).getFlow().getAffinityGroupId();
else  {
    flowId = yFlow.getSubFlows(/*sameSwitchOnly=*/true).get(0).getSubFlowId;
}
stateMachine.setMainAffinityFlowId(flowId)
```
The names and avoiding NPEs could be missing from snippet above, but I hope the idea is clear.